pkg/image/prune/imagepruner.go

```diff
@@ -488,10 +557,33 @@ func (p *imagePruner) Run(pruneImage ImagePruneFunc, pruneStream ImageStreamPrun
 	allNodes := p.g.NodeList()
 
 	imageNodes := imageNodeSubgraph(allNodes)
+	if len(imageNodes) == 0 {
```

Comment:
> When the ImagePruner is created via NewImagePruner, it calls addImagesToGraph, which only adds images managed by our registry.

That's going to make the edge builders for pruning look different than most of the other.  #3404  introduces the idea of creating a synthetic node to fill the graph.  A node that isn't backed by anything real, but that is logically referenced by some other node.  That allows building a graph that can describe its incompleteness and make proposals for resolution.  I think you'd want to handle such a case.